src/main/java/de/tum/in/www1/artemis/service/ExamService.java

```diff
@@ -689,6 +686,15 @@ public void setNumberOfRegisteredUsersForExams(List<Exam> exams) {
         exams.forEach(exam -> exam.setNumberOfRegisteredUsers(registeredUsersCountMap.get(exam.getId()).longValue()));
     }
 
+    /**
+     * Sets the transient attribute numberOfGeneratedStudentExams for the given exam
+     * @param exam Exam for which to compute and set the number of generated student exams
+     */
+    public void setNumberOfGeneratedStudentExams(Exam exam) {
+        Exam tmpExam = findWithStudentExams(exam.getId());
+        exam.setNumberOfGeneratedStudentExams((long) tmpExam.getStudentExams().size());
```

Comment:
please write a query that retrieves the number from the database directly, so that not all the student exams need to be loaded into Java objects

Reply:
Done